File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> sumMap;
        for (int i=0;i<nums.size();i++) {
            if (sumMap.find(target - nums[i]) != sumMap.end()) {
                return {i, sumMap[target-nums[i]]};
            }
            sumMap[nums[i]] = i;
        }
        return {0, 0};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> m = new HashMap<>();
        int n = nums.length;
        int []result = new int[2];
        
        for (int i=0;i<n;i++) {
            if (m.getOrDefault(target - nums[i], -1) == -1) {
                m.put(nums[i], i);
            } else {
                result[0] = i;
                result[1] = m.get(target - nums[i]);
                break;
            }
        }
        
        return result;
    }
}